File: server/main/models/post.js
const mongoose = require('mongoose');

const postSchema = new mongoose.Schema({
    title: {
        type: String,
        required: true,
        trim: true
    },
    body: {
        type: String,
        required: true,
        trim: true,
        default: ''
    },
    image: {
        type: String,
        default: null
    },
    likes: {
        type: [String],
        default: []
    },
    dislikes: {
        type: [String],
        default: []
    },
    comments: {
        type: [
            {
                username: {
                    type: String,
                    required: true,
                },
                body: {
                    type: String,
                    required: true,
                },
                createdAt: {
                    type: Date,
                    default: Date.now
                },
                updatedAt: {
                    type: Date,
                    default: Date.now
                }
            }
        ],
    }
}, {
    timestamps: true,
});

const Post = mongoose.model('Post', postSchema);

module.exports = Post;

File: server/usersManager/models/post.py
from mongoengine import Document, EmbeddedDocument, StringField, ListField, EmbeddedDocumentField, DateTimeField


class Comment(EmbeddedDocument):
    _id = ObjectIdField(default=lambda: ObjectId(), unique=True)
    username = StringField(required=True)
    body = StringField(required=True)
    createdAt = DateTimeField(default=datetime.now)
    updatedAt = DateTimeField(default=datetime.now)

class Post(Document):
    username = StringField(required=True, trim=True)
    title = StringField(required=True, trim=True)
    body = StringField(required=True, trim=True, default='')
    image = StringField(default=None)
    likes = ListField(StringField(), default=[])
    dislikes = ListField(StringField(), default=[])
    comments = ListField(EmbeddedDocumentField(Comment), default=[])

    meta = {'indexes': [
        {'fields': ['comments._id'], 'unique': True},
        {'fields': ['comments.createdAt']},
        {'fields': ['username']},
        {'indexes': ['createdAt']},
    ]}

